plugins/workspace-minimap/src/positioned_minimap.ts

```diff
@@ -14,23 +14,131 @@ import {Minimap} from './minimap';
 
 /**
  * A positionable version of minimap that implements IPositionable.
- * @param {Minimap} minimap The minimap to position.
- * @implements {Blockly.IPositionable}
  */
-export class PositionedMinimap {
+export class PositionedMinimap implements Blockly.IPositionable {
     protected minimap: Minimap;
+    protected primaryWorkspace: Blockly.WorkspaceSvg;
+    protected margin: number;
+    protected top: number;
+    protected left: number;
+    protected width: number;
+    protected height: number;
+    id: string;
     /**
-     * Constructor for a minimap
-     * @param {Minimap} minimap The workspace to sit in.
+     * Constructor for a positionable minimap.
+     * @param workspace The workspace to mirror.
      */
-    constructor(minimap: Minimap) {
-      this.minimap = minimap;
+    constructor(workspace: Blockly.WorkspaceSvg) {
+      this.primaryWorkspace = workspace;
+      this.minimap = new Minimap(this.primaryWorkspace);
+      this.id = 'minimap';
+      this.margin = 20;
+      this.top = 0;
+      this.left = 0;
+      this.width = 225; // TODO: Bumping size responsiveness to future branch.
+      this.height = 150;
     }
 
     /**
      * Initialize.
      */
     init(): void {
-    //   this.work
+      this.primaryWorkspace.getComponentManager().addComponent({
+        component: this,
+        weight: 3,
+        capabilities: [Blockly.ComponentManager.Capability.POSITIONABLE],
+      });
+      this.minimap.init();
+      this.setMinimapCss();
+      this.primaryWorkspace.resize();
```

Comment:
Is this call necessary?

Reply:
The setMinimapCss is not, the other two are.

Reply:
What is the `primaryWorkspace.resize()` doing? I wouldn't expect adding a minimap to affect the size of the primary workspace.

Reply:
You're right in that adding a minimap does not affect the size of the primary workspace, but the resize function doesn't only resize the workspace. It also deals with positioning. "This should be called when something changes that requires recalculating dimensions and positions of the trash, zoom, toolbox, etc."

Without this the minimap does not show up.